specs/WindowControlsOverlayConfiguration.md

```diff
@@ -0,0 +1,156 @@
+WebView2 Window Controls
+===
+
+# Background
+This API allows you to enable and configure the Webview2 Window Controls Overlay.
+The Overlay is a region on the top right/left of the webview window which contains 
+the caption buttons (minimize, maximize, restore, close). Enabing the Overlay allows
+for custom app title bars rendered completely inside the webview window.
+
+This API is designed to be used in addition with the other non-client region APIs 
+and features. These include `app-region: drag`, and `IsNonClientRegionSupportEnabled`.
+
+# Conceptual pages (How To)
+Here is a concept doc on the window controls overlay: https://wicg.github.io/window-controls-overlay/#concepts. 
+This was written for the PWA counterpart  for this feature. From the perspective of
+HTML & Javascript layers, everything there applies in Webview2 as well.
+
+# Examples
+
+## Win32 C++
+```cpp 
+AppWindow::AppWindow() {
+    m_mainWindow = CreateWindowExW(
+        WS_EX_CONTROLPARENT, 
+        GetWindowClass(),
+        L""
+        WS_POPUPWINDOW, 
+        0,0, 800, 800, 
+        nullptr, nullptr, 
+        g_hInstance, nullptr);
+}
+
+void  AppWindow::OnCreateWebview2ControllerCompleted(HRESULT hr, ICoreWebview2Controller* controller) 
+{
+
+    wil::com_ptr<ICoreWebView2> coreWebView2;
+    CHECK_FAILURE(m_controller->get_CoreWebView2(&coreWebView2));
+
+    wil::com_ptr<ICoreWebView2> coreWebView2_28;
+    CHECK_FAILURE(coreWebView2->QueryInterface(&coreWebView2_28));
+
+    wil::com_ptr<ICoreWebView2WindowControlsOverlaySettings> windowControlsOverlaySettings;
+    CHECK_FAILURE(coreWebView2_28->get_WindowControlsOverlaySettings(&wco_config));
+
+    CHECK_FAILURE(wco_config->put_IsEnabled(true));
+    COREWEBVIEW2_COLOR color {1, 0, 0, 225};
+    CHECK_FAILURE(wco_config->put_TitleBarBackgroundColor(color));
+}
+```
+## .NET C#
+```c#
+// WebView2 control is defined in the xaml
+// <wv2:WebView2 x:Name="webView" Source="https://www.microsoft.com/"/>
+public MainWindow() 
+{
+    InitializeComponent();
+    m_AppWindow.TitleBar.ExtendsContentIntoTitleBar = true;
+
+    CoreWebView2WindowControlsOverlaySettings config = Webview2.CoreWebivew2.WindowControlsOverlaySettings;
+    config.IsEnabled = true; 
+    config.color = Color.FromARGB(0, 0, 255);
```

Comment:
Does the order of setting these properties matter?  If I set enabled=true before setting .color might there be a frame rendered of the overlay with the default color?

Reply:
Please update docs to say when does the IsEnabled change take affect - immediate or batched once the UI thread has a chance to work.

Prefer: the order shouldn't matter. If order matters we'll follow up.